SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+        if ( [string]::IsNullOrEmpty($sidstr) ) {
+            Write-Message -Level Verbose -Message "Account not found!" -ForegroundColor Red
+            #exit -1
+        }
+
+        Write-Message -Level Verbose -Message "Account SID: $($sidstr)" -ForegroundColor DarkCyan
+        $tmp = ""
+        $tmp = [System.IO.Path]::GetTempFileName()
+        Write-Message -Level Verbose -Message "Export current Local Security Policy" -ForegroundColor DarkCyan
+        secedit.exe /export /cfg "$($tmp)" 
+        $c = ""
+        $c = Get-Content -Path $tmp
+        $currentSetting = ""
+        foreach ($s in $c) {
+            if ( $s -like "SeManageVolumePrivilege*") {
+                $x = $s.split("=", [System.StringSplitOptions]::RemoveEmptyEntries)
+                $currentSetting = $x[1].Trim()
+            }
+        }
+
+
+        if ( $currentSetting -notlike "*$($sidstr)*" ) {
+            Write-Message -Level Verbose -Message "Modify Setting ""Perform Volume Maintenance Task""" -ForegroundColor DarkCyan
+       
+            if ( [string]::IsNullOrEmpty($currentSetting) ) {
+                $currentSetting = "*$($sidstr)"
+            }
+            else {
+                $currentSetting = "*$($sidstr),$($currentSetting)"
+            }
+       
+            Write-Message -Level Verbose -Message "$currentSetting"
+       
+            $outfile = @"
+[Unicode]
+Unicode=yes
+[Version]
+signature="`$CHICAGO`$"
+Revision=1
+[Privilege Rights]
+SeManageVolumePrivilege = $($currentSetting)
+"@
+       
+            $tmp2 = ""
+            $tmp2 = [System.IO.Path]::GetTempFileName()
+       
+       
+            Write-Message -Level Verbose -Message "Import new settings to Local Security Policy" -ForegroundColor DarkCyan
+            $outfile | Set-Content -Path $tmp2 -Encoding Unicode -Force
+            #notepad.exe $tmp2
+            Push-Location (Split-Path $tmp2)
+       
+            try {
+                secedit.exe /configure /db "secedit.sdb" /cfg "$($tmp2)" /areas USER_RIGHTS 
+                #Write-Message -Level Verbose -Message "secedit.exe /configure /db ""secedit.sdb"" /cfg ""$($tmp2)"" /areas USER_RIGHTS "
+            }
+            finally {  
+                Pop-Location
+            }
+        }
+        else {
+            Write-Message -Level Verbose -Message "NO ACTIONS REQUIRED! Account already in ""Perform Volume Maintenance Task""" -ForegroundColor DarkCyan
+        }
+        Write-Message -Level Verbose -Message "Done." -ForegroundColor DarkCyan 
+    }
+
+    
+
+    #Now configure the right amount of TempDB files.
+
+    $val = 1
+
+    WHILE ($val -ne $NumberOfCores) {
+        $sqlM = 'ALTER DATABASE tempdb ADD FILE ( NAME = N''tempdev' + $val + ''', FILENAME = N''' + $TempVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\DATA\tempdev' + $val + '.ndf'' , SIZE = 64MB , FILEGROWTH = 64MB)'
```

Comment:
We may have a command for this as well. Please check `Set-DbaTempdbConfig`

Reply:
We'll have to discuss if this is in scope, however.

Reply:
As this is a config option from SQL 2016 up as well, i believe it is in scope, for the same reasons the perform volume maintenance setting should be in scope. At least for 2016 and up and again as a service for the earlier versions.

Reply:
agreed, great points